### ISSUE TITLE
CVE-2021-44906  & CVE-2022-3517 vulnerabilities found in  angular-devkit/build-angular related to wildcard package

### ISSUE THREAD

**Issue by TharunSB** [angular/angular-cli#24836]
### Command

other

### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Intro
We are using [trivy](https://www.aquasec.com/products/trivy/) to scan vulnerable dependencies in our angular projects and we have found two critical vulnerabilities from wildcard@2.0.0 package which is part of **@angular-devkit/build-angular/15.2.2**. 
 Dependency tree is as follows

@angular-devkit/build-angular/15.2.2
>           ----- webpack-merge@5.8.0
>                     -----wildcard@2.0.0
>                              -----minimist@1.2.0
>                              -----minimatch@3.0.4

### CVE details:

1. minimist@1.2.0 
- [CVE-2021-44906  ](https://avd.aquasec.com/nvd/2021/cve-2021-44906/)
2. minimatch@3.0.4
- [CVE-2022-3517](https://avd.aquasec.com/nvd/2022/cve-2022-3517/)

### Suggestion for fix:
trivy also suggested that vulnerability is fixed in minimist@1.2.6 and  minimatch@3.0.5. since there is no recent updates in wildcard package, alternative to be found for wildcard.

### Minimal Reproduction

In any angular project with **@angular-devkit/build-angular/15.2.2**, we can run **trivy** scanner using the following command given trivy is installed in your local.

> trivy fs . --security-checks vuln --severity critical,high

reports generated on terminal will show wildcard package with two vulnerabilities with respect to minimist and minimatch

### Your Environment

```text
Angular CLI: 15.2.2
Node: 14.20.0
Package Manager: pnpm 7.17.1
OS: darwin x64

Angular: 15.2.2
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, language-service, localize, material
... platform-browser, platform-browser-dynamic, platform-server
... router, service-worker

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1102.9
@angular-devkit/build-angular   15.2.2
@angular-devkit/core            11.2.8
@angular-devkit/schematics      11.2.8
@angular/pwa                    0.1102.8
@nguniversal/builders           15.1.0
@nguniversal/express-engine     15.1.0
@schematics/angular             11.2.8
rxjs                            6.6.7
typescript                      4.9.4
```

### Anything else relevant?

This vulnerability is observed on **@angular-devkit/build-angular/12.2.18** and above versions. Haven't verified for angular version below 12.

**Comment by clydin** [comment]
The `wildcard@2.0.0` package does not appear to have any dependencies:
https://unpkg.com/browse/wildcard@2.0.0/package.json
https://www.npmjs.com/package/wildcard?activeTab=dependencies

**Comment by TharunSB** [comment]
@clydin  vulnerabilities are coming from dev dependencies of wildcard package.
[wildcard](https://unpkg.com/browse/wildcard@2.0.0/package.json ) has "tape": "^4.6.3"  in its dev dependency.
further we can see

  ----- wildcard@2.0.0

         -----tape@4.6.3
                  -----minimist@1.2.0 
                  -----glob@7.1.1
                          ------minimatch@3.0.2


[tape](https://unpkg.com/browse/tape@4.6.3/package.json) has minimist as direct dependency which is vulnerable package and minimatch is a dependency from [glob](https://unpkg.com/browse/glob@7.1.1/package.json) package  inside tape

**Comment by clydin** [comment]
`devDependencies` of transitive dependencies are not installed by package managers.
This can be shown by executing `npm ls tape` in a newly created Angular v15.2.2 project:
```
└── (empty)
```

The output of `npm ls minimist` for the same project is:
```
└─┬ karma@6.4.1
  └─┬ mkdirp@0.5.6
    └── minimist@1.2.8
```
The output of `npm ls minimatch` for the same project is:
```
├─┬ @angular-devkit/build-angular@15.2.2
│ ├─┬ babel-plugin-istanbul@6.1.1
│ │ └─┬ test-exclude@6.0.0
│ │   ├─┬ glob@7.2.3
│ │   │ └── minimatch@3.1.2 deduped
│ │   └── minimatch@3.1.2
│ └─┬ glob@8.1.0
│   └── minimatch@5.1.6
├─┬ @angular/cli@15.2.2
│ └─┬ pacote@15.1.0
│   ├─┬ @npmcli/run-script@6.0.0
│   │ └─┬ node-gyp@9.3.1
│   │   └─┬ glob@7.2.3
│   │     └── minimatch@3.1.2
│   ├─┬ npm-packlist@7.0.4
│   │ └─┬ ignore-walk@6.0.1
│   │   └── minimatch@6.2.0
│   └─┬ sigstore@1.1.0
│     └─┬ tuf-js@1.1.1
│       └─┬ @tufjs/models@1.0.0
│         └── minimatch@6.2.0
├─┬ karma-coverage@2.2.0
│ └── minimatch@3.1.2
└─┬ karma@6.4.1
  ├─┬ glob@7.2.3
  │ └── minimatch@3.1.2 deduped
  ├── minimatch@3.1.2
  └─┬ rimraf@3.0.2
    └─┬ glob@7.2.3
      └── minimatch@3.1.2
```

As shown above, all package versions are above the requested minimum versions.

**Comment by clydin** [comment]
Additionally, manually installing `tape` directly (`npm install tape@4.6.3`) results in the following package versions with `minimist@1.2.8` and `minimatch@3.1.2` both above the suggested version threshold:
```
└─┬ tape@4.6.3
  ├── deep-equal@1.0.1
  ├── defined@1.0.1
  ├─┬ for-each@0.3.3
  │ └── is-callable@1.2.7
  ├── function-bind@1.1.1
  ├─┬ glob@7.1.7
  │ ├── fs.realpath@1.0.0
  │ ├─┬ inflight@1.0.6
  │ │ ├── once@1.4.0 deduped
  │ │ └── wrappy@1.0.2
  │ ├── inherits@2.0.4 deduped
  │ ├─┬ minimatch@3.1.2
  │ │ └─┬ brace-expansion@1.1.11
  │ │   ├── balanced-match@1.0.2
  │ │   └── concat-map@0.0.1
  │ ├─┬ once@1.4.0
  │ │ └── wrappy@1.0.2 deduped
  │ └── path-is-absolute@1.0.1
  ├─┬ has@1.0.3
  │ └── function-bind@1.1.1 deduped
  ├── inherits@2.0.4
  ├── minimist@1.2.8
  ├── object-inspect@1.2.2
  ├── resolve@1.1.7
  ├─┬ resumer@0.0.0
  │ └── through@2.3.8 deduped
  ├─┬ string.prototype.trim@1.1.2
  │ ├─┬ define-properties@1.2.0
  │ │ ├─┬ has-property-descriptors@1.0.0
  │ │ │ └── get-intrinsic@1.2.0 deduped
  │ │ └── object-keys@1.1.1
  │ ├─┬ es-abstract@1.21.2
  │ │ ├─┬ array-buffer-byte-length@1.0.0
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ └── is-array-buffer@3.0.2 deduped
  │ │ ├── available-typed-arrays@1.0.5
  │ │ ├─┬ call-bind@1.0.2
  │ │ │ ├── function-bind@1.1.1 deduped
  │ │ │ └── get-intrinsic@1.2.0 deduped
  │ │ ├─┬ es-set-tostringtag@2.0.1
  │ │ │ ├── get-intrinsic@1.2.0 deduped
  │ │ │ ├─┬ has-tostringtag@1.0.0
  │ │ │ │ └── has-symbols@1.0.3 deduped
  │ │ │ └── has@1.0.3 deduped
  │ │ ├─┬ es-to-primitive@1.2.1
  │ │ │ ├── is-callable@1.2.7 deduped
  │ │ │ ├─┬ is-date-object@1.0.5
  │ │ │ │ └── has-tostringtag@1.0.0 deduped
  │ │ │ └─┬ is-symbol@1.0.4
  │ │ │   └── has-symbols@1.0.3 deduped
  │ │ ├─┬ function.prototype.name@1.1.5
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ ├── define-properties@1.2.0 deduped
  │ │ │ ├── es-abstract@1.21.2 deduped
  │ │ │ └── functions-have-names@1.2.3
  │ │ ├─┬ get-intrinsic@1.2.0
  │ │ │ ├── function-bind@1.1.1 deduped
  │ │ │ ├── has-symbols@1.0.3 deduped
  │ │ │ └── has@1.0.3 deduped
  │ │ ├─┬ get-symbol-description@1.0.0
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ └── get-intrinsic@1.2.0 deduped
  │ │ ├─┬ globalthis@1.0.3
  │ │ │ └── define-properties@1.2.0 deduped
  │ │ ├─┬ gopd@1.0.1
  │ │ │ └── get-intrinsic@1.2.0 deduped
  │ │ ├── has-property-descriptors@1.0.0 deduped
  │ │ ├── has-proto@1.0.1
  │ │ ├── has-symbols@1.0.3
  │ │ ├── has@1.0.3 deduped
  │ │ ├─┬ internal-slot@1.0.5
  │ │ │ ├── get-intrinsic@1.2.0 deduped
  │ │ │ ├── has@1.0.3 deduped
  │ │ │ └─┬ side-channel@1.0.4
  │ │ │   ├── call-bind@1.0.2 deduped
  │ │ │   ├── get-intrinsic@1.2.0 deduped
  │ │ │   └── object-inspect@1.12.3
  │ │ ├─┬ is-array-buffer@3.0.2
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ ├── get-intrinsic@1.2.0 deduped
  │ │ │ └── is-typed-array@1.1.10 deduped
  │ │ ├── is-callable@1.2.7 deduped
  │ │ ├── is-negative-zero@2.0.2
  │ │ ├─┬ is-regex@1.1.4
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ └── has-tostringtag@1.0.0 deduped
  │ │ ├─┬ is-shared-array-buffer@1.0.2
  │ │ │ └── call-bind@1.0.2 deduped
  │ │ ├─┬ is-string@1.0.7
  │ │ │ └── has-tostringtag@1.0.0 deduped
  │ │ ├─┬ is-typed-array@1.1.10
  │ │ │ ├── available-typed-arrays@1.0.5 deduped
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ ├── for-each@0.3.3 deduped
  │ │ │ ├── gopd@1.0.1 deduped
  │ │ │ └── has-tostringtag@1.0.0 deduped
  │ │ ├─┬ is-weakref@1.0.2
  │ │ │ └── call-bind@1.0.2 deduped
  │ │ ├── object-inspect@1.12.3
  │ │ ├── object-keys@1.1.1 deduped
  │ │ ├─┬ object.assign@4.1.4
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ ├── define-properties@1.2.0 deduped
  │ │ │ ├── has-symbols@1.0.3 deduped
  │ │ │ └── object-keys@1.1.1 deduped
  │ │ ├─┬ regexp.prototype.flags@1.4.3
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ ├── define-properties@1.2.0 deduped
  │ │ │ └── functions-have-names@1.2.3 deduped
  │ │ ├─┬ safe-regex-test@1.0.0
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ ├── get-intrinsic@1.2.0 deduped
  │ │ │ └── is-regex@1.1.4 deduped
  │ │ ├─┬ string.prototype.trim@1.2.7
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ ├── define-properties@1.2.0 deduped
  │ │ │ └── es-abstract@1.21.2 deduped
  │ │ ├─┬ string.prototype.trimend@1.0.6
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ ├── define-properties@1.2.0 deduped
  │ │ │ └── es-abstract@1.21.2 deduped
  │ │ ├─┬ string.prototype.trimstart@1.0.6
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ ├── define-properties@1.2.0 deduped
  │ │ │ └── es-abstract@1.21.2 deduped
  │ │ ├─┬ typed-array-length@1.0.4
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ ├── for-each@0.3.3 deduped
  │ │ │ └── is-typed-array@1.1.10 deduped
  │ │ ├─┬ unbox-primitive@1.0.2
  │ │ │ ├── call-bind@1.0.2 deduped
  │ │ │ ├── has-bigints@1.0.2
  │ │ │ ├── has-symbols@1.0.3 deduped
  │ │ │ └─┬ which-boxed-primitive@1.0.2
  │ │ │   ├─┬ is-bigint@1.0.4
  │ │ │   │ └── has-bigints@1.0.2 deduped
  │ │ │   ├─┬ is-boolean-object@1.1.2
  │ │ │   │ ├── call-bind@1.0.2 deduped
  │ │ │   │ └── has-tostringtag@1.0.0 deduped
  │ │ │   ├─┬ is-number-object@1.0.7
  │ │ │   │ └── has-tostringtag@1.0.0 deduped
  │ │ │   ├── is-string@1.0.7 deduped
  │ │ │   └── is-symbol@1.0.4 deduped
  │ │ └─┬ which-typed-array@1.1.9
  │ │   ├── available-typed-arrays@1.0.5 deduped
  │ │   ├── call-bind@1.0.2 deduped
  │ │   ├── for-each@0.3.3 deduped
  │ │   ├── gopd@1.0.1 deduped
  │ │   ├── has-tostringtag@1.0.0 deduped
  │ │   └── is-typed-array@1.1.10 deduped
  │ └── function-bind@1.1.1 deduped
  └── through@2.3.8
```

**Comment by alan-agius4** [comment]
Closing as per above.
